src/core/Accordion/index.tsx

```diff
@@ -0,0 +1,31 @@
+import { AccordionProps as RawAccordionProps } from "@material-ui/core";
+import React from "react";
+import { ExtraProps, StyledAccordion } from "./style";
+
+export type AccordionProps = RawAccordionProps & ExtraProps;
+
+const Accordion = (props: AccordionProps) => {
+  const { children, useDivider, togglePosition = "right", id } = props;
+  const [expanded, setExpanded] = React.useState<string | false>(false);
+
+  const handleChange =
+    (panel: string) =>
+    (_event: React.ChangeEvent<unknown>, isExpanded: boolean) => {
+      setExpanded(isExpanded ? panel : false);
+    };
+  return (
+    <StyledAccordion
+      id={id}
+      square
+      useDivider={useDivider}
+      togglePosition={togglePosition}
+      expanded={expanded === id}
+      onChange={handleChange(id)}
```

Comment:
should this be `() => handleChange(id)`? Otherwise it only executes once, right?

Reply:
onChange is already a callback function, so we're passing the function that it's going to call. If this was just a regular react component, you're correct, but MUI accounts for the callback